packages/internal-test-helpers/lib/test-cases/application.ts

```diff
@@ -3,15 +3,26 @@ import Application from '@ember/application';
 import { Router } from '@ember/-internals/routing';
 
 import { runTask, runLoopSettled } from '../run';
+import Resolver from '../test-resolver';
+import { assert as emberAssert } from '@ember/debug';
+import Controller from '@ember/controller';
+import ApplicationInstance from '@ember/application/instance';
 
-export default class ApplicationTestCase extends TestResolverApplicationTestCase {
-  constructor() {
-    super(...arguments);
+export default abstract class ApplicationTestCase extends TestResolverApplicationTestCase {
+  application: Application;
+  applicationInstance?: ApplicationInstance;
+  resolver: Resolver;
+
+  constructor(assert: QUnit['assert']) {
+    super(assert);
 
     let { applicationOptions } = this;
     this.application = runTask(this.createApplication.bind(this, applicationOptions));
 
-    this.resolver = this.application.__registry__.resolver;
+    // TODO: Review this cast
+    let resolver = this.application.__registry__.resolver;
+    emberAssert('expected a resolver', resolver instanceof Resolver);
+    this.resolver = resolver;
 
     if (this.resolver) {
       this.resolver.add('router:main', Router.extend(this.routerOptions));
```

Comment:
Given the changes above, I think this is now a meaningless check, since it wouldn't pass the `emberAssert` if there were no resolver?